### ISSUE TITLE
Ansible-based operator logging verbosity flags don't affect events written to stdout

### ISSUE THREAD

**Issue by FNaughton** [operator-framework/operator-sdk#4446]
## Bug Report

I'm attempting to lower the log level of my ansible-based operator for every day use. Logs are collected from stdout of the operator pod. 
I would also like to be able to increase log levels for investigation when there's an issue.

#### What did you do?

I followed the guidelines for adjusting log levels [here](https://v0-19-x.sdk.operatorframework.io/docs/ansible/reference/advanced_options/#ansible-verbosity)

Before any change I ran `ps -ef` inside the container and see the anible-runner start with command:
` /usr/bin/python3.6 /usr/local/bin/ansible-runner -vv --rotate-artifacts 20 -p /opt/ansible/playbook.yml -i ....`

With the annotation in the watched CR
```
annotations:
    "ansible.operator-sdk/verbosity": "0"
```
I see ( -vv gone from command )
` /usr/bin/python3.6 /usr/local/bin/ansible-runner --rotate-artifacts 20 -p /opt/ansible/playbook.yml -i  .....`

#### What did you expect to see?

I expect to see the log verbosity at stdout is the same as what is written to internal log files
`/tmp/ansible-operator/runner/<group>/<version>/<kind>/<namespace>/<resource>/artifacts/latest/stdout`

#### What did you see instead? Under which circumstances?

Log levels written to stdout remain at info level regardless of the annotation.
The annotation in the CR affected the verbosity of logs written to file, but not what is written to stdout.

#### Environment

**Operator type:**

 /language ansible 

**Kubernetes cluster type:**

Openshift 4.5.17

`$ operator-sdk version`

ansible-operator:v0.19.0

`$ go version` (if language is Go)

`$ kubectl version`

v1.18.3+45b9524

**Comment by dalbani** [comment]
I happen to have run into this very same issue a few of days ago.
And pulled my hair for a couple of hours trying to understand why Ansible logging doesn't work correctly.
It boils down to how the Ansible operator program handles logs from Ansible (Runner).
See https://github.com/operator-framework/operator-sdk/blob/master/internal/ansible/events/log_events.go#L50.
By default, the output from Ansible is only included when:
-  `e.Event == eventapi.EventPlaybookOnTaskStart && !setFactAction && !debugAction`
- `e.Event == eventapi.EventRunnerOnOk && debugAction`

Which I find more than restrictive and strange. Why give the `debug` action any more precedence over the rest?
And why is `set_fact` excluded?
Furthermore, there's also an missing event called `runner_item_on_ok` which is sent by Ansible Runner in the case of a `loop`.

Like @FNaughton, I expect the Ansible operator to forward all the output from Ansible (Runner), leaving the choice to the user how many information needs to be gathered, based on the existing `verbosity` setting.
Basically what is contained in the `stdout` property of all the JSON files in `/tmp/ansible-operator/runner/<group>/<version>/<kind>/<namespace>/<resource>/artifacts/latest/job_events/`.

By the way, why set the default Ansible verbosity level to 2 (i.e. `-vv`) in https://github.com/operator-framework/operator-sdk/blob/master/internal/ansible/flags/flag.go#L66?
A value of 0 seems to me a more sensible option.

**Comment by stiller-leser** [comment]
I too was expecting `ANSIBLE_STDOUT_CALLBACK=actionable` to have an effect.

**Comment by nelsonpraveen** [comment]
The ansible runner inside the operator seems to have the STDOUT_CALLBACK hardcoded inside it.

bash-4.4$ grep STDOUT_CALLBACK /usr/local/lib/python3.6/site-packages/ansible_runner/runner_config.py
            self.env['ANSIBLE_STDOUT_CALLBACK'] = 'minimal'
            self.env['ANSIBLE_STDOUT_CALLBACK'] = 'awx_display'

Is there a way to configure the CALLBACK plugin (to say oneline) ?

**Comment by fridim** [comment]
Hi, any known workaround until this is fixed ?

**Comment by FNaughton** [comment]
@fabianvf - tried adding the envvar 
```
env:
        - name: ANSIBLE_DEBUG_LOGS
          value: "True"
```
It does indeed print the ansible run to the console. But this is _in addition_ to the messages logged by default. I would like to control the verbosity of these default messages. Our logging solution collects from stdout.
On the plus side, the debug logs respect the verbosity set in the watched CR annotation. Perhaps we could disable the default message?

**Comment by FNaughton** [comment]
Testing with an image built from operator-sdk v1.9 I still see info level logs at stdout even with log level set to 0.
Wondering if I'm missing something here?
Dockerfile:
```
FROM mylocalrepo/8-jre-rhel:latest as builder
FROM quay.io/operator-framework/ansible-operator:v1.9.0
...
```

Annotation in watched CR:
```
annotations:
    ansible.sdk.operatorframework.io/verbosity: "0"
```

**Comment by VenkatRamaraju** [comment]
Hi @FNaughton,

Can you show us the current info logs you are getting vs the type of output you are expecting?

With the fix from the PR above, the stdout logs associated with a task will be dumped out (based on annotation/env variable verbosity) during that task's execution. However, we still left the logs that are outputted during the initialization of the operator. Are these the info logs you are referring to?

**Comment by VenkatRamaraju** [comment]
Hi @FNaughton,

We have started looking into this again. Here are some clarification points:
- We have two types of logging packages at use - [Zap](https://pkg.go.dev/go.uber.org/zap) and [Log](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log).
- The fix I submitted in the [PR above](https://github.com/operator-framework/operator-sdk/pull/4961) dumps StdOut logs to the console when verbosity is set as an annotation or environment variable. Keep in mind this PR limits Ansible **task**-related logs only, not the initialization logs that you reference in your [comment](https://github.com/operator-framework/operator-sdk/issues/4446#issuecomment-869516898).

Reading your use case, you might want to use `--zap-log-level=error` CLI flag to only see the Ansible task output with no logs, such as below:
```
$ ansible-operator run local --zap-log-level=error

--------------------------- Ansible Task StdOut -------------------------------

TASK [Gathering Facts] *********************************************************

-------------------------------------------------------------------------------

--------------------------- Ansible Task StdOut -------------------------------

 TASK [Hello World] ********************************
ok: [localhost] => {
    "msg": "Hello, World!"
}

-------------------------------------------------------------------------------

----- Ansible Task Status Event StdOut (cache.example.com/v1alpha1, Kind=Memcached, memcached-sample/default) -----


PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0


----------
```

I am also submitting a PR soon that allows _even less_ output while running the operator, using a CLI flag `--ansible-log-events`, that will look like the following:
```
$ ansible-operator run local --zap-log-level=error --ansible-log-events=nothing

----- Ansible Task Status Event StdOut (cache.example.com/v1alpha1, Kind=Memcached, memcached-sample/default) -----


PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0


----------
```
Feel free to weigh in and let us know if this is what you are looking for and any other suggestions you have as well.  Thanks!
